File: 5- Temperatura/vb/TemperaturaVB/TemperaturaVB/Form1.vb
Public Class frmTemperatura




    Private Sub btnOk_Click(sender As Object, e As EventArgs) Handles btnOk.Click
        Dim tempEntrada, tempSalida As Double

        If (txtIn.Text IsNot String.Empty) And Not txtIn.Text = "." And Not txtIn.Text = "-" And Not txtIn.Text = ".-" And Not txtIn.Text = "-." And Not txtIn.Text.Contains(".-") Then



            tempEntrada = Val(txtIn.Text)

            If radinC.Checked Then



                If radOutF.Checked Then

                    tempSalida = tempEntrada * 1.8 + 32
                    txtOut.Text = tempSalida

                ElseIf radOutK.Checked Then

                    If tempEntrada >= (-273.15) Then

                        tempSalida = tempEntrada + 273.15
                        txtOut.Text = tempSalida

                    ElseIf tempEntrada < (-273.15) Then

                        txtOut.Text = 0
                        MsgBox("La temperatura no puede ser menor a 0 Kelvin")

                    End If


                ElseIf radOutR.Checked Then

                    If tempEntrada >= (-273.15) Then

                        tempSalida = tempEntrada * 1.8 + 491.67
                        txtOut.Text = tempSalida

                    ElseIf tempEntrada < (-273.15) Then

                        txtOut.Text = 0
                        MsgBox("La temperatura no puede ser menor a 0 Rankine")

                    End If



                End If

            ElseIf radInF.Checked Then

                If radOutC.Checked Then

                    tempSalida = (tempEntrada - 32) / 1.8
                    txtOut.Text = tempSalida

                ElseIf radOutK.Checked Then

                    If tempEntrada >= (-459.67) Then

                        tempSalida = ((tempEntrada - 32) / 1.8) + 273.15
                        txtOut.Text = tempSalida

                    ElseIf tempEntrada < (-459.67) Then

                        txtOut.Text = 0
                        MsgBox("La temperatura no puede ser menor a 0 Kelvin")

                    End If



                ElseIf radOutR.Checked Then

                    If tempEntrada >= (-459.67) Then

                        tempSalida = tempEntrada + 459.67
                        txtOut.Text = tempSalida

                    ElseIf tempEntrada < (-459.67) Then

                        txtOut.Text = 0
                        MsgBox("La temperatura no puede ser menor a 0 Rankine")

                    End If



                End If

            ElseIf radInK.Checked Then

                If tempEntrada < 0 Then

                    MsgBox("La temperatura no puede ser menor a 0 Kelvin")



                ElseIf radOutC.Checked Then

                    tempSalida = (tempEntrada - 273.15)
                    txtOut.Text = tempSalida

                ElseIf radOutF.Checked Then

                    tempSalida = ((tempEntrada - 273.15) * 1.8) + 32
                    txtOut.Text = tempSalida

                ElseIf radOutR.Checked Then

                    tempSalida = tempEntrada * 1.8
                    txtOut.Text = tempSalida

                End If

            ElseIf radInR.Checked Then

                If tempEntrada < 0 Then

                    MsgBox("La temperatura no puede ser menor a 0 Rankine")

                ElseIf radOutC.Checked Then

                    tempSalida = (tempEntrada - 491.67) * (5 / 9)
                    txtOut.Text = tempSalida

                ElseIf radOutF.Checked Then

                    tempSalida = ((tempEntrada - 459.67))
                    txtOut.Text = tempSalida

                ElseIf radOutK.Checked Then

                    tempSalida = tempEntrada * (5 / 9)
                    txtOut.Text = tempSalida

                End If

            End If
        End If

    End Sub

    Private Sub radinC_CheckedChanged(sender As Object, e As EventArgs) Handles radinC.CheckedChanged

        If radinC.Checked Then

            radOutC.Enabled = False
            radOutF.Checked = True

        Else

            radOutC.Enabled = True

        End If

    End Sub

    Private Sub radInF_CheckedChanged(sender As Object, e As EventArgs) Handles radInF.CheckedChanged

        If radInF.Checked Then

            radOutF.Enabled = False
            radOutK.Checked = True

        Else

            radOutF.Enabled = True

        End If

    End Sub

    Private Sub radInK_CheckedChanged(sender As Object, e As EventArgs) Handles radInK.CheckedChanged

        If radInK.Checked Then

            radOutK.Enabled = False
            radOutR.Checked = True

        Else

            radOutK.Enabled = True

        End If

    End Sub

    Private Sub radInR_CheckedChanged(sender As Object, e As EventArgs) Handles radInR.CheckedChanged

        If radInR.Checked Then

            radOutR.Enabled = False
            radOutC.Checked = True

        Else

            radOutR.Enabled = True

        End If

    End Sub

    Private Sub txtIn_KeyPress(ByVal sender As Object, ByVal e As System.Windows.Forms.KeyPressEventArgs) Handles txtIn.KeyPress


        If txtIn.Text = String.Empty Then

            e.Handled = Not Char.IsDigit(e.KeyChar) And Not Char.IsControl(e.KeyChar) And Not e.KeyChar = "." And Not e.KeyChar = "-"

        ElseIf Not txtIn.Text.Contains(".") Then
            e.Handled = Not Char.IsDigit(e.KeyChar) And Not Char.IsControl(e.KeyChar) And Not e.KeyChar = "."
        ElseIf Not txtIn.Text.Contains("-") Then
            e.Handled = Not Char.IsDigit(e.KeyChar) And Not Char.IsControl(e.KeyChar) And Not e.KeyChar = "-"
        ElseIf txtIn.Text.Contains(".") And txtIn.Text.Contains("-") Then
            e.Handled = Not Char.IsDigit(e.KeyChar) And Not Char.IsControl(e.KeyChar)
        End If






    End Sub
End Class

File: 5- Temperatura/cs/TemperaturaCS/TemperaturaCS/Form1.cs
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace TemperaturaCS
{
	public partial class frmTemperaturaCS : Form
	{
		public frmTemperaturaCS()
		{
			InitializeComponent();
		}

		public void btnOk_Click(object sender, EventArgs e)
		{
			
			double tempEntrada, tempSalida;
			if (txtIn.Text != string.Empty & !((txtIn.Text ?? "") == ".") & !((txtIn.Text ?? "") == "-") & !((txtIn.Text ?? "") == ".-") & !((txtIn.Text ?? "") == "-.") & !txtIn.Text.Contains(".-"))
			{
				tempEntrada = Conversion.Val(txtIn.Text);
				if (radinC.Checked)
				{
					if (radOutF.Checked)
					{
						tempSalida = tempEntrada * 1.8 + 32;
						txtOut.Text = Conversions.ToString(tempSalida);
					}
					else if (radOutK.Checked)
					{
						if (tempEntrada >= -273.15)
						{
							tempSalida = tempEntrada + 273.15;
							txtOut.Text = Conversions.ToString(tempSalida);
						}
						else if (tempEntrada < -273.15)
						{
							txtOut.Text = Conversions.ToString(0);
							MessageBox.Show("La temperatura no puede ser menor a 0 Kelvin");
						}
					}
					else if (radOutR.Checked)
					{
						if (tempEntrada >= -273.15)
						{
							tempSalida = tempEntrada * 1.8 + 491.67;
							txtOut.Text = Conversions.ToString(tempSalida);
						}
						else if (tempEntrada < -273.15)
						{
							txtOut.Text = Conversions.ToString(0);
							MessageBox.Show("La temperatura no puede ser menor a 0 Rankine");
						}
					}
				}
				else if (radInF.Checked)
				{
					if (radOutC.Checked)
					{
						tempSalida = (tempEntrada - 32) / 1.8;
						txtOut.Text = Conversions.ToString(tempSalida);
					}
					else if (radOutK.Checked)
					{
						if (tempEntrada >= -459.67)
						{
							tempSalida = (tempEntrada - 32) / 1.8 + 273.15;
							txtOut.Text = Conversions.ToString(tempSalida);
						}
						else if (tempEntrada < -459.67)
						{
							txtOut.Text = Conversions.ToString(0);
							MessageBox.Show("La temperatura no puede ser menor a 0 Kelvin");
						}
					}
					else if (radOutR.Checked)
					{
						if (tempEntrada >= -459.67)
						{
							tempSalida = tempEntrada + 459.67;
							txtOut.Text = Conversions.ToString(tempSalida);
						}
						else if (tempEntrada < -459.67)
						{
							txtOut.Text = Conversions.ToString(0);
							MessageBox.Show("La temperatura no puede ser menor a 0 Rankine");
						}
					}
				}
				else if (radInK.Checked)
				{
					if (tempEntrada < 0)
					{
						MessageBox.Show("La temperatura no puede ser menor a 0 Kelvin");
					}
					else if (radOutC.Checked)
					{
						tempSalida = tempEntrada - 273.15;
						txtOut.Text = Conversions.ToString(tempSalida);
					}
					else if (radOutF.Checked)
					{
						tempSalida = (tempEntrada - 273.15) * 1.8 + 32;
						txtOut.Text = Conversions.ToString(tempSalida);
					}
					else if (radOutR.Checked)
					{
						tempSalida = tempEntrada * 1.8;
						txtOut.Text = Conversions.ToString(tempSalida);
					}
				}
				else if (radInR.Checked)
				{
					if (tempEntrada < 0)
					{
						MessageBox.Show("La temperatura no puede ser menor a 0 Rankine");
					}
					else if (radOutC.Checked)
					{
						tempSalida = (tempEntrada - 491.67) * (5 / (double)9);
						txtOut.Text = Conversions.ToString(tempSalida);
					}
					else if (radOutF.Checked)
					{
						tempSalida = tempEntrada - 459.67;
						txtOut.Text = Conversions.ToString(tempSalida);
					}
					else if (radOutK.Checked)
					{
						tempSalida = tempEntrada * (5 / (double)9);
						txtOut.Text = Conversions.ToString(tempSalida);
					}
				}
			}

		}






		public void radinC_CheckedChanged(object sender, EventArgs e)
		{

			if (radinC.Checked)
			{
				radOutC.Enabled = false;
				radOutF.Checked = true;
			}
			else
			{
				radOutC.Enabled = true;
			}

		}



		public void radInF_CheckedChanged(object sender, EventArgs e)
		{

			if (radInF.Checked)
			{
				radOutF.Enabled = false;
				radOutK.Checked = true;
			}
			else
			{
				radOutF.Enabled = true;
			}

		}



		public void radInK_CheckedChanged(object sender, EventArgs e)
		{

			if (radInK.Checked)
			{
				radOutK.Enabled = false;
				radOutR.Checked = true;
			}
			else
			{
				radOutK.Enabled = true;
			}


		}





		public void radInR_CheckedChanged(object sender, EventArgs e)
		{

			if (radInR.Checked)
			{
				radOutR.Enabled = false;
				radOutC.Checked = true;
			}
			else
			{
				radOutR.Enabled = true;
			}


		}








		public void txtIn_KeyPress(object sender, KeyPressEventArgs e)
		{


			if ((txtIn.Text ?? "") == (string.Empty ?? ""))
			{
				e.Handled = !char.IsDigit(e.KeyChar) & !char.IsControl(e.KeyChar) & !(Conversions.ToString(e.KeyChar) == ".") & !(Conversions.ToString(e.KeyChar) == "-");
			}
			else if (!txtIn.Text.Contains("."))
			{
				e.Handled = !char.IsDigit(e.KeyChar) & !char.IsControl(e.KeyChar) & !(Conversions.ToString(e.KeyChar) == ".");
			}
			else if (!txtIn.Text.Contains("-"))
			{
				e.Handled = !char.IsDigit(e.KeyChar) & !char.IsControl(e.KeyChar) & !(Conversions.ToString(e.KeyChar) == "-");
			}
			else if (txtIn.Text.Contains(".") & txtIn.Text.Contains("-"))
			{
				e.Handled = !char.IsDigit(e.KeyChar) & !char.IsControl(e.KeyChar);
			}

		}

		
	}
}
